keps/sig-architecture/3138-increase-reliability-bar/README.md

```diff
@@ -0,0 +1,415 @@
+
+In order to achieve the goals across the whole project and all SIGs, we need
+to ensure that the principles will not be ignored. So while we hope that building
+the appropriate culture and appropriate encouragement will be enough, we also
+define the enforcement mechanisms that could be used if everything else fails.
+
+
+## Motivation
+
+As Kubernetes is becoming more and more mature, and widely used in production,
+we need to increase the quality bar to reduce the amount of outages and issues
+our users are facing.
+
+While companies/cloud-providers do not generally publish the data about outages,
+we already have a lot of data that proves the above. The last [Production
+Readiness Survey][] (performed by Production Readiness subproject of SIG
+Architecture) proved that there are tens of examples where users had to rollback
+their cluster due to issues with examples of many different reasons, including:
+broken feature, failing components, scalability issues or even failing clusters.
+Being forced to rollback clearly means a significant reliability issue.
+There are also many examples described in different blog posts or Kubecon talks,
+e.g. a recent [How to Break your Cluster with networking][]).
+However, many issues that were discussed even years ago during other Kubecon
+talks (e.g. [1][] or [2][]) also weren’t addressed. And these aren’t just
+hypothetical - many of us have seen them in real production.
+
+However, it’s not just about those huge outages. It’s also about smaller incidents
+(some of them even self-recovering, but still causing issues in the meantime).
+By having a better testing discipline we could catch a lot of them. But what’s
+even worse, we aren’t catching some even though they are exposed by our tests,
+because they are simply getting lost/oversight amongst a huge number of test
+flakes. Networking is a great example - back then around 1.20-1.21 releases
+their tests were extremely flaky. Thanks to a huge and organized effort by the
+SIG network they reduced their flakiness by orders of magnitude. While many of the
+problems were test-related, addressing them allowed them to uncover real system
+issues, like [this example][], which must have affected many real production
+clusters before.
+
+To make Kubernetes (or any other project) sustainable, we should ensure that
+existing architecture, individual features, tests etc. really works as expected.
+If there is work needed to improve reliability (or stability) of those, it
+should come before new features.
+This is often not the case currently. We significantly rely on the heroic
+efforts of very few individuals that are keeping us on the surface with new bugs,
+issues, test flakes etc. This approach isn’t sustainable and doesn’t scale.
+We need to make it a business of everyone in the whole community.
+
+[Production Readines Survey]: https://datastudio.google.com/reporting/2e9c7439-202b-48a9-8c57-4459e0d69c8d/page/GC5HB
```

Comment:
can we open this up access to this dashboard somehow? I still can't access it.

Reply:
@deads2k @johnbelamaric - will you be able to open it?